### PR TITLE
Load variables from file.

### PR DESCRIPTION
**Pull request by mithunbharadwaj**
**Overview:**
Introduce _load_variables_from_json() in JinjaFormatterService to load variables from JSON. Includes unit tests for robustness.

**Changes:**

- Added _load_variables_from_json() for JSON variable loading.
- Implemented unit tests

### CODE REVIEW

**Review by ateska — APPROVED** [review]

File: asabiris/formatter/jinja/service.py [hunk 1]
```diff
@@ -26,6 +28,35 @@ def __init__(self, app, service_name="JinjaService"):
...
 
 		self.Environment = jinja2.Environment()
+		self._load_variables_from_json()
+
+	def _load_variables_from_json(self):
+		"""
+		Load variables from a JSON file specified in the configuration.
+		Overwrites any existing variables from the configuration file with the same keys.
+		"""
+		json_path_str = asab.Config.get('jinja', 'variables', fallback=None)
```
Comment (ateska) [inline asabiris/formatter/jinja/service.py:38]: `fallback=None` is implicit

Comment (mithunbharadwaj) [reply asabiris/formatter/jinja/service.py:38]: @ateska I always do this for explicitness and clarity while reading the code.

File: asabiris/formatter/jinja/service.py [hunk 2]
```diff
@@ -26,6 +28,35 @@ def __init__(self, app, service_name="JinjaService"):
...
+		if not json_path_str:
+			L.info("No JSON file specified for variables in configuration.")
```
Comment (ateska) [inline asabiris/formatter/jinja/service.py:40]: `if json_path_str is not None:`

You don't need to print anything ...

**Review by ateska — CHANGES_REQUESTED** [review]

File: asabiris/formatter/jinja/service.py (hunk 1 above)
Comment (ateska) [reply asabiris/formatter/jinja/service.py:38]: It increases a cognitive load. `.get()` in Python implicitly returns None then item is not found.

**Review by ateska — APPROVED** [review]

File: asabiris/formatter/jinja/service.py [hunk 3]
```diff
@@ -26,6 +28,45 @@ def __init__(self, app, service_name="JinjaService"):
...
+				json_data = json.load(json_file)
+		except IOError as io_err:
+			L.warning("Failed to read JSON file '{}': {}".format(json_path, io_err))
+			return
+		except json.JSONDecodeError as json_err:
+			L.warning("Invalid JSON format in file '{}': {}".format(json_path, json_err))
+			return
+
+		self.Variables.update(json_data)
+		L.info("Variables successfully loaded from JSON file '{}'.".format(json_path))
```
Comment (ateska) [inline asabiris/formatter/jinja/service.py:69]: `L.debug()`

**Review by ateska — CHANGES_REQUESTED** [review]
Last remark, then mergeable.

**Pull request merged**